### ISSUE TITLE
how to make it to show in a grid

### ISSUE THREAD

**Issue by jjhesk** [nex3z/ToggleButtonGroup#14]
how is that possible to support grid layout?

**Comment by nex3z** [comment]
SingleSelectToggleGroup or MultiSelectToggleGroup can be used directly in GridLayout:

```xml
<?xml version="1.0" encoding="utf-8"?>
<android.support.v7.widget.GridLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:padding="8dp"
    tools:context="com.nex3z.togglebuttongroup.sample.SingleSelectActivity">

    <com.nex3z.togglebuttongroup.SingleSelectToggleGroup
        android:id="@+id/group_choices"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        app:tbgCheckedButton="@+id/choice_a">

        <com.nex3z.togglebuttongroup.button.CircularToggle
            android:id="@+id/choice_a"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:textSize="16sp"
            android:text="A"/>

        <com.nex3z.togglebuttongroup.button.CircularToggle
            android:id="@+id/choice_b"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:textSize="16sp"
            android:text="B"/>

        <com.nex3z.togglebuttongroup.button.CircularToggle
            android:id="@+id/choice_c"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:textSize="16sp"
            android:text="C"/>

        <com.nex3z.togglebuttongroup.button.CircularToggle
            android:id="@+id/choice_d"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"
            android:textSize="16sp"
            android:text="D"/>

    </com.nex3z.togglebuttongroup.SingleSelectToggleGroup>

</android.support.v7.widget.GridLayout>
```

![screenshot_2017-11-14-16-52-40-513_com nex3z togglebuttongroup sample](https://user-images.githubusercontent.com/5645004/32771768-c5b892f6-c95e-11e7-9ccf-02e6a67594c5.png)

In the layout above, SingleSelectToggleGroup is responsible for the layout of CircularToggles, not the GridLayout.

You can also put toggle buttons (eg. CircularToggle) in the GridLayout directly:
```xml
<?xml version="1.0" encoding="utf-8"?>
<android.support.v7.widget.GridLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:padding="8dp"
    app:columnCount="2"
    tools:context="com.nex3z.togglebuttongroup.sample.SingleSelectActivity">

    <com.nex3z.togglebuttongroup.button.CircularToggle
        android:id="@+id/choice_a"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:textSize="16sp"
        android:text="A"/>

    <com.nex3z.togglebuttongroup.button.CircularToggle
        android:id="@+id/choice_b"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:textSize="16sp"
        android:text="B"/>

    <com.nex3z.togglebuttongroup.button.CircularToggle
        android:id="@+id/choice_c"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:textSize="16sp"
        android:text="C"/>

    <com.nex3z.togglebuttongroup.button.CircularToggle
        android:id="@+id/choice_d"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:textSize="16sp"
        android:text="D"/>

</android.support.v7.widget.GridLayout>
```

![screenshot_2017-11-14-16-53-01-861_com nex3z togglebuttongroup sample](https://user-images.githubusercontent.com/5645004/32771774-c9bc289a-c95e-11e7-8e8d-85db5604affc.png)

In the layout above, GridLayout is responsible for the layout of the CircularToggles. But without the help of SingleSelectToggleGroup or MultiSelectToggleGroup, you will have to manage CircularToggles' checked state all by yourself.

If your goal is to implement a button grid, [here is one possible solution](https://github.com/nex3z/ToggleButtonGroup/issues/12#issuecomment-325923001).

**Comment by jjhesk** [comment]
great job! @nex3z I have found that this isnt the best solutions since I need to have all the functionalities from multiselects and toggles. I would take this and fork with and other library based flexbox-layout developed by google. it seems like a good foundations for all the layout managements. I finally managed to resolve the issue by having the library to merge with flexbox-layout foundation. its now in my fork.
